### ISSUE TITLE
Install MySQL database

### ISSUE THREAD

**Issue by Norien** [Norien/Home-Assistant-Config#11]
Amazing documentation here by @brianjking

https://github.com/brianjking/homeassistant-config/blob/master/docs/mysql.md

Also saw in Git chat @dale3h is releasing a community video on this topic.

**Comment by Norien** [comment]
Have lots of errors regarding install.

$ sudo apt --fix-broken install
`grep: /etc/mysql/: No such file or directory
Failed to stop mysql.service: Unit mysql.service not loaded.
invoke-rc.d: initscript mysql, action "stop" failed.
invoke-rc.d returned 5
There is a MySQL server running, but we failed in our attempts to stop it.
Stop it yourself and try again!
dpkg: error processing archive /var/cache/apt/archives/mysql-server-5.7_5.7.18-0ubuntu0.17.04.1_amd64.deb (--unpack):
 subprocess new pre-installation script returned error exit status 1
Errors were encountered while processing:
 /var/cache/apt/archives/mysql-server-5.7_5.7.18-0ubuntu0.17.04.1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
`

$ sudo apt-get update && sudo apt-get upgrade
```
Get:1 http://security.ubuntu.com/ubuntu zesty-security InRelease [89.2 kB]
Hit:2 http://ca.archive.ubuntu.com/ubuntu zesty InRelease
Get:3 http://ca.archive.ubuntu.com/ubuntu zesty-updates InRelease [89.2 kB]
Hit:4 http://ca.archive.ubuntu.com/ubuntu zesty-backports InRelease
Fetched 178 kB in 1s (149 kB/s)
Reading package lists... Done
Reading package lists... Done
Building dependency tree
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 mysql-server : Depends: mysql-server-5.7 but it is not installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
```

sudo apt-get install mysql-server-5.7
```
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages were automatically installed and are no longer required:
  galera-3 libjemalloc1 mariadb-common socat
Use 'sudo apt autoremove' to remove them.
Suggested packages:
  tinyca
The following NEW packages will be installed:
  mysql-server-5.7
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
5 not fully installed or removed.
Need to get 0 B/3,358 kB of archives.
After this operation, 50.0 MB of additional disk space will be used.
Preconfiguring packages ...
(Reading database ... 176284 files and directories currently installed.)
Preparing to unpack .../mysql-server-5.7_5.7.18-0ubuntu0.17.04.1_amd64.deb ...
grep: /etc/mysql/: No such file or directory
Failed to stop mysql.service: Unit mysql.service not loaded.
invoke-rc.d: initscript mysql, action "stop" failed.
invoke-rc.d returned 5
There is a MySQL server running, but we failed in our attempts to stop it.
Stop it yourself and try again!
dpkg: error processing archive /var/cache/apt/archives/mysql-server-5.7_5.7.18-0ubuntu0.17.04.1_amd64.deb (--unpack):
 subprocess new pre-installation script returned error exit status 1
Errors were encountered while processing:
 /var/cache/apt/archives/mysql-server-5.7_5.7.18-0ubuntu0.17.04.1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```
- $ sudo rm /etc/systemd/system/mysqld.service
- $ sudo rm /etc/systemd/system/mysql.service
- $ sudo reboot

**Comment by Norien** [comment]
These are the steps I took to correct this:

Back up your `my.cnf` file in `/etc/mysql` and remove it
Remove the folder `/etc/mysql/mysql.conf.d/`
Verify you don't have a `my.cnf` file stashed somewhere else (I did in my home dir!) or in `/etc/alternatives/my.cnf`.
Backup and remove `/etc/mysql/debian.cnf` files (not sure if needed, but just in case)
`sudo apt purge mysql-server mysql-server-5.7 mysql-server-core-5.7`
`sudo apt install mysql-server`
In case your syslog shows an error like "mysqld: Can't read dir of '/etc/mysql/conf.d/'" create a symbolic link ln -s /etc/mysql/mysql.conf.d /etc/mysql/conf.d Then the service should be able to start with service mysql start.
That got it working!
